pyproject.toml

```diff
@@ -28,3 +63,29 @@ addopts =[
     "--ruff",
     "--ruff-format",
 ]
+
+[dependency-groups]
+deploy = [
```

Comment:
This is a bit of a nitpick, but our pyproject.toml has 2 categories of thing in it:

- Package management stuff (`project`, `dependency-groups`, `tool.uv.<things>`)
- Config for unrelated tools that happen to live in pyproject.toml (`tool.ruff`, `tool.pytest`)

Could we group all the package management stuff together in the file seperately from the unrelated config just for readability. At the moment it is a bit of a random mashup.